client/component-lab/Notification/NotificationView.coffee

```diff
@@ -0,0 +1,207 @@
+_ = require 'lodash'
+React = require 'app/react'
+ReactDOM = require 'react-dom'
+kd = require 'kd'
+
+getVendorTransition = require 'app/util/getVendorTransition'
+
+Button = require '../Button/Button'
+Box = require '../Box/Box'
+
+styles = require './Notification.stylus'
+Helpers = require './helpers'
+Constants = require './constants'
+
+
+module.exports = class NotificationView extends React.Component
+
+    constructor: (props) ->
+
+      super props
+      @_style = { zIndex: 100 + @props.index }
+      @_notificationTimer = null
+      @_isMounted = no
+      @_removeCount = 0
+      @state =
+        removed : no
+      @_hideNotification = @_hideNotification.bind this
+      @_removeNotification = @_removeNotification.bind this
+      @_onTransitionEnd = @_onTransitionEnd.bind this
+
+
+    _hideNotification: ->
+
+      @_notificationTimer.clear() if @_notificationTimer
+      @setState { removed : yes } if @_isMounted
+      @_removeNotification()
+
+
+    _removeNotification: ->
+
+      @props.onRemove(@props.notification.uid)
+
+
+    _dismiss: ->
+
+      @_hideNotification() if @props.notification.dismissible
+
+
+    _onTransitionEnd: ->
+
+      if @_removeCount == 0 and @state.removed
+        @_removeCount++
+        @_removeNotification()
+
+
+    autoDismissible: (notification) ->
+
+      not notification.dismissible and not notification.primaryButtonTitle and not notification.secondaryButtonTitle
+
+
+    componentDidMount: ->
+
+      self = this
+      transitionEvent = getVendorTransition()
+      element = ReactDOM.findDOMNode(self)
+      notification = @props.notification
+      @_isMounted = yes
+      if transitionEvent
+        element.addEventListener transitionEvent, @_onTransitionEnd
+      if @autoDismissible(notification)
+        @_notificationTimer = new (Helpers.Timer)((->
+          self._hideNotification()
+          return
+        ), notification.duration)
+
+
+    _handleMouseEnter: ->
+
+      if @autoDismissible(@props.notification)
```

Comment:
i did not comment on every occurrence, please take another look at your changes and apply these comments to wherever it applies.